source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -22,6 +22,8 @@
 namespace Envoy {
 namespace TcpProxy {
 
+const std::string PerConnectionCluster::KEY = "envoy.tcp_proxy.cluster";
```

Comment:
I think Google had problems with defines conflicts with short ALL_CAPS variable names, perhaps make this `Key`?